features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/App.js

```diff
@@ -95,21 +78,29 @@ class Protected extends Component {
                 </BaseLayout>
             );
         }
+
         let params = qs.stringify({referrer: this.props.location.pathname});
-        if (this.state.authConfigs) {
-            if (this.state.authConfigs.is_sso_enabled) {
-                var authorizationEndpoint = this.state.authConfigs.authorizationEndpoint;
-                var client_id = this.state.authConfigs.client_id;
-                var callback_URL = this.state.authConfigs.callback_url;
-                var scopes = this.state.authConfigs.scopes;
-                window.location = authorizationEndpoint + "?response_type=code&client_id=" + client_id + "&redirect_uri=" + callback_URL + "&scope=" + scopes;
+        return (
+            <Redirect to={{pathname: '/login', search: params}}/>
+        );
+
+        if(this.state.updateSSO) {
+            this.state.updateSSO = false;
+
+            if (this.state.authConfigs) {
+                if (this.state.authConfigs.is_sso_enabled.value) {
+                    const authorizationEndpoint = this.state.authConfigs.authorizationEndpoint.value;
+                    const client_id = this.state.authConfigs.client_id.value;
+                    const callback_URL = `${this.state.authConfigs.callback_url.value}`;
+                    const scopes = this.state.authConfigs.scopes.value;
+                    window.location = `${authorizationEndpoint}?response_type=code&client_id=${client_id}` +
+                        `&redirect_uri=${callback_URL}&scope=${scopes}`;
+                } else {
```

Comment:
Empty else block?

Reply:
Deleted the whole block. This was moved to Login.js. Fixed with https://github.com/wso2/carbon-apimgt/pull/4843/commits/a828722b82ca00020f3c34576767cf25ec19757e